### DISCUSSION TITLE
`Add Development Container` command should be available in Codespaces

### DISCUSSION THREAD

**Discussion by alefragnani** [community/community#26836] [Codespaces]
<p>The <em>zero config</em> in Codespaces is great, and works perfectly, congrats! :clap:</p>
<p>But, for those that use VS Code on a daily basis, sometimes it’s interesting to reuse those pre-defined <code>.devcontainer</code> resources, and be able run the same <em>config</em> locally.</p>
<p>The Codespaces documentation has a <a href="https://docs.github.com/en/free-pro-team@latest/github/developing-online-with-codespaces/configuring-codespaces-for-your-project#using-a-pre-built-container-configuration" rel="noopener nofollow ugc">Use a pre-built container configuraiton</a> section, which describes how to to add those <code>.devcontainer</code> resources on you repo (clone, copy, add) <em>manually</em>. But why not use the same <code>Add Development Container</code> command from the <strong>Remote Development</strong> extension, directly from inside Codespaces?</p>
<p>Doing so, no matter where you <em>start the remote experience</em>. You could have exactly the same experience.</p>
<p>Thank you</p>

**Comment by Chuxel** [comment, marked as answer]
<p>Thanks for the feature request! That is in fact the plan! The “rebuild” workflow is also planned.</p>

**Comment by jkeech** [comment]
<p>We are working on this right now :slight_smile:. We’re setting it up so that both the VS Code - Remote Containers and Codespaces extensions can share that functionality. It should be ready in the next couple weeks.</p>

**Comment by alefragnani** [comment]
<p>That’s great to hear!</p>
<p>Thank you for your great work :clap:</p>
